src/utils.c

```diff
@@ -2436,7 +2463,12 @@ void utils_start_new_geany_instance(const gchar *doc_path)
 GEANY_API_SYMBOL
 gchar *utils_get_real_path(const gchar *file_name)
 {
-	return tm_get_real_path(file_name);
+	gchar *path = NULL;
+
+	if (file_name)
+		path = realpath(file_name, NULL);
```

Comment:
We don't really need to implement a wrapper for `realpath()` if the only caller always passes NULL.

I suggest this
``` 
gchar *path = NULL;

if (file_name)
{
#if defined(G_OS_WIN32) && !defined(HAVE_REALPATH)
    gsize size = GetFullPathNameA(pathname, 0, NULL, NULL);
    path = g_new0(char, size);
    GetFullPathNameA(pathname, size, path, NULL);
#else
    path = realpath(file_name, NULL);
#endif
}
return path;
```

Reply:
Agree with @kugel-, the fatally broken `realpath()` has left the building, the projects where I have noticed all now assume the [POSIX 2008](https://pubs.opengroup.org/onlinepubs/9699919799/functions/realpath.html) behaviour with NULL, nobody tries to allocate themselves, so the only choice is between Windows and POSIX.